lib/spack/spack/cmd/ci.py

```diff
@@ -360,10 +359,10 @@ def ci_rebuild(args):
             mirror_msg = "artifact buildcache enabled, mirror url: {0}".format(pipeline_mirror_url)
             tty.debug(mirror_msg)
 
-    # Whatever form of root_spec we got, use it to get a map giving us concrete
-    # specs for this job and all of its dependencies.
-    spec_map = spack_ci.get_concrete_specs(env, root_spec, job_spec_pkg_name, compiler_action)
-    job_spec = spec_map[job_spec_pkg_name]
+    # Get the concrete spec to be built by this job.  We're using all 32 chars
+    # of the hash, so the list will just have one entry, unless there are hash
+    # collisions.
+    job_spec = env.get_by_hash(job_spec_dag_hash)[0]
```

Comment:
Can you turn this comment into a hard requirement please? You're obtaining `job_spec_dag_hash` from some environment variable `SPACK_JOB_SPEC_DAG_HASH` so it's only true by convention and maybe the assumption breaks after a refactor, so it feels brittle.

The database has `query_one` for example, which asserts that the number of matches in <= 1. Something like that should be added to the environment.